ui/pages/multichain-accounts/multichain-accounts-connect-page/multichain-accounts-connect-page.tsx

```diff
@@ -0,0 +1,605 @@
+
+export const MultichainAccountsConnectPage: React.FC<
+  MultichainConnectPageProps
+> = ({
+  request,
+  permissionsRequestId,
+  rejectPermissionsRequest,
+  approveConnection,
+  targetSubjectMetadata,
+}) => {
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
+  const [pageMode, setPageMode] = useState<MultichainAccountsConnectPageMode>(
+    MultichainAccountsConnectPageMode.Summary,
+  );
+
+  const existingPermissions = useSelector((state) =>
+    getPermissions(state, request.metadata?.origin),
+  );
+
+  const existingCaip25CaveatValue = useMemo(
+    () =>
+      existingPermissions
+        ? getCaip25CaveatValueFromPermissions(existingPermissions)
+        : null,
+    [existingPermissions],
+  );
+
+  const requestedCaip25CaveatValue = getCaip25CaveatValueFromPermissions(
+    request.permissions,
+  );
+
+  const requestedCaip25CaveatValueWithExistingPermissions =
+    existingCaip25CaveatValue
+      ? mergeCaip25CaveatValues(
+          requestedCaip25CaveatValue,
+          existingCaip25CaveatValue,
+        )
+      : requestedCaip25CaveatValue;
+
+  const requestedCaipAccountIds = getCaipAccountIdsFromCaip25CaveatValue(
+    requestedCaip25CaveatValue,
+  );
+
+  const requestedNamespaces = getAllNamespacesFromCaip25CaveatValue(
+    requestedCaip25CaveatValueWithExistingPermissions,
+  );
+
+  const requestedNamespacesWithoutWallet = requestedNamespaces.filter(
+    (namespace) => namespace !== KnownCaipNamespace.Wallet,
+  );
+
+  const networkConfigurationsByCaipChainId = useSelector(
+    getAllNetworkConfigurationsByCaipChainId,
+  );
+
+  const [nonTestNetworkConfigurations, testNetworkConfigurations] = useMemo(
+    () =>
+      Object.entries(networkConfigurationsByCaipChainId).reduce(
+        ([nonTestNetworksList, testNetworksList], [chainId, network]) => {
+          const caipChainId = chainId as CaipChainId;
+          const isTestNetwork =
+            CAIP_FORMATTED_EVM_TEST_CHAINS.includes(caipChainId);
+          (isTestNetwork ? testNetworksList : nonTestNetworksList).push({
+            ...network,
+            caipChainId,
+          });
+          return [nonTestNetworksList, testNetworksList];
+        },
+        [
+          [] as EvmAndMultichainNetworkConfigurationsWithCaipChainId[],
+          [] as EvmAndMultichainNetworkConfigurationsWithCaipChainId[],
+        ],
+      ),
+    [networkConfigurationsByCaipChainId],
+  );
+
+  const requestedCaipChainIds = getAllScopesFromCaip25CaveatValue(
+    requestedCaip25CaveatValueWithExistingPermissions,
+  ).filter((chainId) => {
+    const { namespace } = parseCaipChainId(chainId);
+    return namespace !== KnownCaipNamespace.Wallet;
+  });
+
+  const requestedCaipChainIdsWithDefault = useMemo(
+    () => [
+      ...new Set([
+        ...requestedCaipChainIds,
+        ...nonTestNetworkConfigurations.map(({ caipChainId }) => caipChainId),
+        ...testNetworkConfigurations.map(({ caipChainId }) => caipChainId),
```

Comment:
`requestedCaipChainIds` doesn't need to be in here because `nonTestNetworkConfigurations` and `testNetworkConfigurations` would include all of the wallet's supported caip chain ids already